### PR TITLE
bulk service added, gem refactored

### PR DESCRIPTION
**Pull request by babakgh**
https://forward3d.atlassian.net/browse/PB-380
bulk service added
gem refactored

### CODE REVIEW

File: lib/bing-ads-reporting/ad_insight_service.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            },
+            ns('CustomDateRangeStart') => {
+              ns('Day') => period.from.day,
+              ns('Month') => period.from.month,
+              ns('Year') => period.from.year
+            }
+            # ns('PredefinedTime') => options[:time]
+          }
+        },
+        :attributes! => {ns('ReportRequest') => {
```
Comment (chiting) [inline lib/bing-ads-reporting/ad_insight_service.rb]: Could you re-write this part please?

File: lib/bing-ads-reporting/core_service.rb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                                            ns('AuthenticationToken') => @settings[:authenticationToken] }
+      end
+      Savon.client({
+                    wsdl: wdsl,
+                    log_level: :info,
+                    namespaces: { 'xmlns:arr' => 'http://schemas.microsoft.com/2003/10/Serialization/Arrays',
+                                  'xmlns:i' => "http://www.w3.org/2001/XMLSchema-instance" },
+                    soap_header: header
+                    })
+                    # .merge({pretty_print_xml: true, log_level: :debug, log: true, logger: @logger})) # for more logging
```
Comment (chiting) [inline lib/bing-ads-reporting/core_service.rb]: If we don't need that, can we remove it please?

File: README.md [hunk 3]
```diff
@@ -8,22 +8,46 @@
 
 Initialise service
 
-    service = BingAdsReporting::Service.new({developerToken: '', applicationToken: '', authenticationToken: '', :accountId: '', customerId: ''})
+- Ad Insight Service
+
+```
+    service = BingAdsReporting::AdInsightService.new({developerToken: '', applicationToken: '', authenticationToken: '', :accountId: '', customerId: ''})
```
Comment (chiting) [inline README.md:14]: Maybe for later (or now if you have time) but it looks like both of these classes are coupled with the data structure passed in the argument.

I think for more maintainability we should decouple it into its own class (or struct) as it seems these always come together.

File: lib/bing-ads-reporting/ad_insight_service.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+              ns('Day') => period.to.day,
+              ns('Month') => period.to.month,
+              ns('Year') => period.to.year
+            },
+            ns('CustomDateRangeStart') => {
+              ns('Day') => period.from.day,
+              ns('Month') => period.from.month,
+              ns('Year') => period.from.year
+            }
+            # ns('PredefinedTime') => options[:time]
```
Comment (chiting) [inline lib/bing-ads-reporting/ad_insight_service.rb]: Can we remove this if we don't need it anymore?

File: lib/bing-ads-reporting/bulk_service.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
+require_relative 'core_service'
+
+module BingAdsReporting
+  class BulkService < CoreService
+
```
Comment (chiting) [inline lib/bing-ads-reporting/bulk_service.rb]: empty line

File: lib/bing-ads-reporting/core_service.rb (hunk 2 above)
Comment (robertino) [reply lib/bing-ads-reporting/core_service.rb]: it comes in handy in tough times ;)

File: CHANGELOG.txt [hunk 6]
```diff
@@ -1,3 +1,6 @@
+* 2.0.0
+  BingAdsReporting::Service class renamed to BingAdsReporting::AdInsightService
+  BingAdsReporting::BulkService added
```
Comment (robertino) [inline CHANGELOG.txt:3]: i'd add some new lines for spacing here

**Review by robertino — APPROVED** [review]

File: lib/bing-ads-reporting/core_service.rb (hunk 2 above)
Comment (chiting) [reply lib/bing-ads-reporting/core_service.rb]: yeah agreed but I'd say either remove it or leave it and un-comment as it's logging in debug anyway

Comment (babakgh) [reply lib/bing-ads-reporting/core_service.rb]: removed

File: README.md (hunk 3 above)
Comment (babakgh) [reply README.md:14]: later I think

File: lib/bing-ads-reporting/ad_insight_service.rb (hunk 4 above)
Comment (babakgh) [reply lib/bing-ads-reporting/ad_insight_service.rb]: removed

File: lib/bing-ads-reporting/bulk_service.rb (hunk 5 above)
Comment (babakgh) [reply lib/bing-ads-reporting/bulk_service.rb]: done

File: CHANGELOG.txt (hunk 6 above)
Comment (babakgh) [reply CHANGELOG.txt:3]: done

**Review by chiting — APPROVED** [review]

**Pull request merged**